File: solutions/ruby/resistor-color-duo/2/resistor_color_duo.rb
module ResistorColorDuo

  BANDS = {
    'black'  => 0,
    'brown'  => 1,
    'red'    => 2,
    'orange' => 3,
    'yellow' => 4,
    'green'  => 5,
    'blue'   => 6,
    'violet' => 7,
    'grey'   => 8,
    'white'  => 9
  }

  def self.value(bands)
    band1, band2 = bands

    (BANDS[band1] * 10) + BANDS[band2]
  end

end

File: solutions/typescript/resistor-color-duo/2/resistor-color-duo.ts
export function decodedValue(colors: [Color, Color]): number {
    return COLORS.indexOf(colors[0]) * 10 + COLORS.indexOf(colors[1]);
}

export const COLORS = [
    'black',
    'brown',
    'red',
    'orange',
    'yellow',
    'green',
    'blue',
    'violet',
    'grey',
    'white',
] as const;

export type Color = (typeof COLORS)[number];
